subxt/src/config/refine_params.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! Refining params with values fetched from the chain
+
+use crate::Config;
+
+/// Types implementing [`RefineParams`] can be modified to reflect live information from the chain.
+pub trait RefineParams<T: Config> {
+    /// Refine params to an extrinsic. There is usually some notion of 'the param is already set/unset' in types implementing this trait.
+    /// The refinement should most likely not affect cases where a param is in a 'is already set by the user' state.
+    fn refine(&mut self, _account_nonce: u64, _block_number: u64, _block_hash: T::Hash) {}
```

Comment:
Are there currently other types that substrate/polkadot extensions take for constructing their state?
I'm thinking more about how we could easily extend this in the future without causing a breaking change

Reply:
One thing that we could do to _maybe_ minimise the risk of breaking changes a little is to wrap these into a struct like:

```rust
pub struct RefineData<T: Config> {
    account_nonce: u64,
    block_number: u64,
    block_hash: T::Hash
}

impl <T: Config> RefineData {
    pub fn account_nonce(&self) -> u64 { ... }
    pub fn block_number(&self) -> u64 { ... }
    pub fn block_hash(&self) -> u64 { ... }
}
```

That way, we can add params without changing the existing interface :)